docs/datatypes/func.md

```diff
@@ -0,0 +1,78 @@
+# Func
+
+Func is a wrapper around a `run` function `a => F[B]` where F is a functor. Given that, the Func data type is equipped with the known `map` function, and a `mapK` function to apply natural transformations (from a `F` Func get an `G` Func).
+
+The signature: `Func[F[_], A, B]` refers to an `F` functor, `A` source type and `B` target type (`F[B]`).
+
+If you are familiar with `Kleisli` you can recognize it has a similar signature: `Kleisli[F[_], -A, B]` and `Func[F[_], A, B]`. Well yes, `Func` is a less restrictive data type that wraps around functors, and only provides basic methods `run`, `map`, and `mapK`, while `Kleisli` is strong enough to provide composition, flatMap, and more. We will see a more useful data type just next with `AppFunc`. 
+
+## Quick example
+
+```scala mdoc:silent:nest
+import cats.data.{ Func, AppFunc }
+import cats._
+
+val f: Func[List, Int, String] = Func.func((x: Int) => List(x.toString))
+
+val g: Func[List, Int, Option[String]] = f.map((x: String) => if (x=="0") None else Some(x))
+
+val optToList = new (Option ~> List) {
+  def apply[T](opt: Option[T]): List[T] =
+    opt.toList
+}
+
+// We transform the elements of List, of type 
+// Option[String] to List[String]
+g.map(optToList(_))
+// val res0: cats.data.Func[List,Int,List[String]] = ...
+```
+
+
+
+# AppFunc 
+
+AppFunc extends Func to wrap around a special type of functor: Applicative functors.
+
+With applicative functors we can `compose`, form the `product`, and also `traverse` traversable functors
+
+Signature: `AppFunc[F[_], A, B] extends Func[F, A, B]` 
+
+Now, for the reader familiar with `Kleisli`, we find an even more similar data type. `AppFunc` provides compositions of weaker constraint, allowing to compose `AppFunc[F[_], A, B]` with `AppFunc[G[_], C, A]`.   
```

Comment:
I don't think these Kleisli call outs are going to help people trying to figure out what AppFunc is for. Have you considered having a comparison to Kleisli section at the end of the article?

Reply:
Hey, yes actually I think at the beginning it was at the end but a suggestion was to move it to the beginning. Given that most doubts in the PR introducing the code were around "why do we need this if Kleisli is a thing"

Reply:
Ha, I guess that's down to a matter of audience. For cats newcomers, Kleisli is probably at least as foreign as this.